components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -215,6 +216,37 @@ func (r *NotebookReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, err
 	}
 
+	// Check if annotations for the Notebook instance imply a requirement for notebook restart.
```

Comment:
Can we move this whole logic after this https://github.com/opendatahub-io/kubeflow/blob/491c2777e65b6b7b323131f02f38aeacf1818c2d/components/notebook-controller/controllers/notebook_controller.go#L222

so that this logic , work without any interruption: https://github.com/opendatahub-io/kubeflow/blob/491c2777e65b6b7b323131f02f38aeacf1818c2d/components/notebook-controller/controllers/notebook_controller.go#L218.